src/safeposix/syscalls/fs_calls.rs

```diff
@@ -2835,13 +2873,19 @@ impl Cage {
 
             //Construct desired access bits (i.e. 0777) based on the amode parameter
             let mut newmode: u32 = 0;
+            // if we are checking for execute rights
             if amode & X_OK == X_OK {
+                // update newmode with S_IXUSR flags
                 newmode |= S_IXUSR;
             }
+            // if we are checking for write rights
             if amode & W_OK == W_OK {
+                // update newmode with S_IWUSR flags
                 newmode |= S_IWUSR;
             }
+            // if we are checking for read rights
             if amode & R_OK == R_OK {
+                // update newmode with S_IRUSR flags
                 newmode |= S_IRUSR;
             }
```

Comment:
do we check the F_OK flag?

Reply:
we aren't checking for it explicitly, but we already error out if the file doesn't exist. If you want I can add a check here that will return 0 if amode is F_OK

Reply:
i think just explaining that in the comments should be enough

Reply:
have added the check for it already :)